app/src/main/java/cm/aptoide/pt/analytics/analytics/AnalyticsManager.java

```diff
@@ -46,53 +60,139 @@ public void logEvent(Map<String, Object> data, String eventName, Action action,
     private SessionLogger sessionLogger;
 
+    /**
+     * <p>Start the builder.</p>
+     */
     public Builder() {
       eventLoggers = new HashMap<>();
     }
 
+    /**
+     * <p>Adds an {@link EventLogger} and the respective {@code supportedEvents }. </p>
+     * <p>This {@code eventLogger} will allow to register a service to log any of the {@code
+     * supportedEvents
+     * }.</p>
+     *
+     * <p>If this builder was not started yet (see {@link #Builder()}), a
+     * {@link NullPointerException} will occur.</p>
+     *
+     * @param eventLogger The EventLogger to add.
+     * @param supportedEvents A collection of the possible events associated with the {@code
+     * eventLogger}.
+     *
+     * @return A builder with the added {@code eventLogger} and respective {@code supportedEvents}.
+     *
+     * @see NullPointerException
+     */
     public Builder addLogger(EventLogger eventLogger, Collection<String> supportedEvents) {
       eventLoggers.put(eventLogger, supportedEvents);
       return this;
     }
 
+    /**
+     * <p>Adds a {@link SessionLogger}.</p>
+     * <p>This {@code sessionLogger} will allow to start and stop a Flurry session.</p>
+     *
+     * <p>If this builder was not started yet (see {@link #Builder()}), a
+     * {@link NullPointerException} will occur.</p>
+     *
+     * @param sessionLogger A flurry SessionLogger.
+     *
+     * @return A builder with the added {@code sessionLogger}.
+     *
+     * @see NullPointerException
+     */
     public Builder addSessionLogger(SessionLogger sessionLogger) {
       this.sessionLogger = sessionLogger;
       return this;
     }
 
+    /**
+     * <p>Sets a {@link HttpKnockEventLogger} that will allow to register a
+     * service to log a Knock URL {@code String}.</p>
+     *
+     * <p>If this builder was not started yet (see {@link #Builder()}), a
+     * {@link NullPointerException} will occur.</p>
+     *
+     * @param httpKnockEventLogger The {@code httpKnockEventLogger} to log the events.
+     *
+     * @return A builder with the added {@link HttpKnockEventLogger}.
+     *
+     * @see NullPointerException
+     */
     public Builder setKnockLogger(HttpKnockEventLogger httpKnockEventLogger) {
       this.httpKnockEventLogger = httpKnockEventLogger;
       return this;
     }
 
+    /**
+     * <p>Creates an AnalyticsManager.</p>
+     *
+     * <p>If this builder was not started ( see {@link #Builder()} ), a
+     * {@link NullPointerException} will occur.</p>
+     *
+     * <p>An AnalyticsManager needs a {@link HttpKnockEventLogger}, so if no {@link
+     * HttpKnockEventLogger} was added (see {@link
+     * #setKnockLogger(HttpKnockEventLogger)} ) an IllegalArgumentException will be thrown.</p>
+     *
+     * <p>An AnalyticsManager needs at least one {@link EventLogger}, so if no {@link EventLogger}
+     * were added (see {@link #addLogger(EventLogger, Collection)},
+     * an IllegalArgumentException will be thrown.</p>
```

Comment:
Phrasing. I would suggest separating the exception case, like above. Something to the tune of "An AnalyticsManager needs a {@link HttpKnockEventLogger} (see {@link
+     * #setKnockLogger(HttpKnockEventLogger)}.
If no logger was added (...)"
The same applies to the next one.